multimodal/src/autogluon/multimodal/data/process_multimodal.py

```diff
@@ -0,0 +1,227 @@
+                )
+            self.max_len = min(max_len, self.tokenizer.model_max_length)
+        logger.debug(f"text max length: {self.max_len}")
+
+    @property
+    def text_token_ids_key(self):
+        return f"{self.prefix}_{TEXT_TOKEN_IDS}"
+
+    @property
+    def text_segment_ids_key(self):
+        return f"{self.prefix}_{TEXT_SEGMENT_IDS}"
+
+    @property
+    def text_valid_length_key(self):
+        return f"{self.prefix}_{TEXT_VALID_LENGTH}"
+
+    @property
+    def text_token_word_mapping_key(self):
+        return f"{self.prefix}_{TOKEN_WORD_MAPPING}"
+
+    @property
+    def text_word_offsets_key(self):
+        return f"{self.prefix}_{WORD_OFFSETS}"
+
+    @property
+    def label_key(self):
+        return f"{self.prefix}_{LABEL}"
+
+    def collate_fn(self, text_column_names: Optional[List] = None) -> Dict:
+        """
+        Collate multimodal features into a batch.
+        This function will be used when creating Pytorch DataLoader.
+
+        Returns
+        -------
+        A dictionary containing one model's collator function for multimodal data.
+        """
+        fn = {}
+        if self.prefix == NER:
+            fn.update(
+                {
+                    self.text_token_ids_key: Pad(pad_val=self.tokenizer.pad_token_id),
+                    self.text_valid_length_key: Stack(),
+                    self.text_segment_ids_key: Pad(pad_val=0),
+                    self.text_token_word_mapping_key: Pad(pad_val=0),
+                    self.text_word_offsets_key: Pad(pad_val=0),
+                    self.label_key: Stack(),
+                }
+            )
+        return fn
+
+    @staticmethod
+    def get_pretrained_tokenizer(
+        tokenizer_name: str,
+        checkpoint_name: str,
+    ):
+        """
+        Load the tokenizer for a pre-trained huggingface checkpoint.
+
+        Parameters
+        ----------
+        tokenizer_name
+            The tokenizer type, e.g., "bert", "clip", "electra", and "hf_auto".
+        checkpoint_name
+            Name of a pre-trained checkpoint.
+
+        Returns
+        -------
+        A tokenizer instance.
+        """
+        tokenizer_class = ALL_TOKENIZERS[tokenizer_name]
+        return tokenizer_class.from_pretrained(checkpoint_name)
+
+    def process_ner(
+        self,
+        sample: Dict[str, Union[int, float, list]],
+        is_training: bool,
+    ) -> Dict:
+        """
+        Process one NER sample's label and text features.
+        New rules can be added if necessary.
+
+        Parameters
+        ----------
+        labels
+            One sample may have multiple labels.
+
+        Returns
+        -------
+        A dictionary containing one NER sample's features and/or labels.
+        """
+        ret = {}
+        ner_text = sample[TEXT]
+        if is_training or NER_ANNOTATION in sample:
+            ner_annotation = sample[NER_ANNOTATION]
```

Comment:
If passing the `modality_types` inside the MultiModalProcessor, we could avoid detecting the modalities by the hardcoded keys.

Reply:
resolved [here](https://github.com/awslabs/autogluon/pull/2220/commits/b66cdb2aa7df0ecce0003a941a965c00a8012ce2)